ui/src/VoxettaRecordButton.js

```diff
@@ -16,39 +16,85 @@ limitations under the License. */
 import {AudioRecorder} from './AudioRecorder';
 import {LitElement, html, css} from 'lit-element';
 
+/**
+ * Component responsible for recording and uploading audio files. 
+ */
 export class VoxettaRecordButton extends LitElement {
     static get properties() {
         return {
             isRecording: {type: Boolean},
             audioRecorder: {type: Object},
         };
     }
+
     constructor() {
         super();
         this.isRecording = false;
         this.audioRecorder = new AudioRecorder();
     }
-    render() {  
+
+    render() {
+        this.getUploadUrl();  
```

Comment:
it was requested to move the API logic to another file, but anyways, FYI: the element gets re-rendered every time a property is updated, meaning every time `this.isRecording` is changed, `render()` gets called and therefore this asks for a new upload URL again which isn't ideal
https://lit-element.polymer-project.org/guide/lifecycle

Reply:
Oh I completely overlooked this! This makes sense, I'll avoid this in the future

Reply:
It only triggers a redraw for "LitElement" properties, which are properties listed in `static get properties`